File: ios/Classes/search/model/UnifiedGeocodeResult.m
//
// Created by Yohom Bao on 2018-12-14.
//

#import "UnifiedGeocodeResult.h"
#import "AMapCommonObj.h"
#import "AMapSearchKit.h"
#import "NSArray+Rx.h"


@implementation UnifiedGeocodeResult {

}

- (instancetype)initWithAMapGeocodeSearchResponse:(AMapGeocodeSearchResponse *)response {
    self = [super init];
    if (self) {
        _geocodeAddressList = [response.geocodes map:^(AMapGeocode* geocode) {
           return [[UnifiedGeocodeAddress alloc] initWithAMapGeocode:geocode];
        }];
    }

    return self;
}

@end

@implementation UnifiedGeocodeAddress {

}
- (instancetype)initWithAMapGeocode:(AMapGeocode *)geocode {
    self = [super init];
    if (self) {
        _formatAddress = geocode.formattedAddress;
        _province = geocode.province;
        _city = geocode.city;
        _adcode = geocode.adcode;
        _district = geocode.district;
        _township = geocode.township;
        _neighborhood = geocode.neighborhood;
        _building = geocode.building;
        _level = geocode.level;
        _latLng = geocode.location;
    }

    return self;
}

@end

@implementation UnifiedGeocodeQuery {

}
@end

File: android/src/main/kotlin/me/yohom/amapbase/search/model/UnifiedGeocodeResult.kt
package me.yohom.amapbase.search.model

import com.amap.api.services.geocoder.GeocodeAddress
import com.amap.api.services.geocoder.GeocodeQuery
import com.amap.api.services.geocoder.GeocodeResult
import me.yohom.amapbase.search.toLatLng

class UnifiedGeocodeResult(geocodeResult: GeocodeResult) {
    val geocodeQuery: UnifiedGeocodeQuery = UnifiedGeocodeQuery(geocodeResult.geocodeQuery)
    val geocodeAddressList: List<UnifiedGeocodeAddress> = geocodeResult.geocodeAddressList.map { UnifiedGeocodeAddress(it) }
}

class UnifiedGeocodeAddress(geocodeAddress: GeocodeAddress) {
    val formatAddress: String = geocodeAddress.formatAddress
    val province: String = geocodeAddress.province
    val city: String = geocodeAddress.city
    val district: String = geocodeAddress.district
    val township: String = geocodeAddress.township
    val neighborhood: String = geocodeAddress.neighborhood
    val building: String = geocodeAddress.building
    val adcode: String = geocodeAddress.adcode
    val latLng: LatLng = geocodeAddress.latLonPoint.toLatLng()
    val level: String = geocodeAddress.level
}

class UnifiedGeocodeQuery(geocodeQuery: GeocodeQuery) {
    val locationName: String = geocodeQuery.locationName
    val city: String = geocodeQuery.city
}